desc/transform.py

```diff
@@ -57,7 +57,11 @@ def __init__(
         self._basis = basis
         self._rcond = rcond if rcond is not None else "auto"
 
-        if not (self.grid.NFP == self.basis.NFP) and grid.node_pattern != "custom":
+        if (
+            self.grid.num_zeta > 1
```

Comment:
Unequal NFP between the grid and basis can still cause problems for a single zeta coordinate, if the angle is not 0.

Reply:
good point, ill fix this